### ISSUE TITLE
Unable to Push

### ISSUE THREAD

**Issue by EugeneFoo** [brandon-barker/PushBots.NET#3]
Hi,

When I tried this library, it shows: 

"Cannot deserialize the current JSON object (e.g. {\"name\":\"value\"}) into type 'System.Collections.Generic.List`1[PushBots.NET.Models.Device]' because the type requires a JSON array (e.g. [1,2,3]) to deserialize correctly.\r\nTo fix this error either change the JSON to a JSON array (e.g. [1,2,3]) or change the deserialized type so that it is a normal .NET type (e.g. not a primitive type like integer, not a collection type like an array or List<T>) that can be deserialized from a JSON object. JsonObjectAttribute can also be added to the type to force it to deserialize from a JSON object.\r\nPath 'code', line 1, position 8."}

**Comment by brandon-barker** [comment]
Hi,

Please could you paste the code you're trying so I can try and reproduce the error?

**Comment by EugeneFoo** [comment]
Hi Brandon,

Here's the sample project file for your reference:
http://www.filehosting.org/file/details/532138/PushBotTest.zip

Thanks.

**Comment by brandon-barker** [comment]
Hi Eugene,

First thing I can notice after a quick look is the way you're defining the `Payload` in the `SinglePush` object.

Please could you try changing Line 41 in `PushController.cs` from:

``` c#
Payload = new JObject(""),
```

to

``` c#
Payload = new JObject(),
```

I think the error is that it is trying to parse a blank string ("") as a JSON object.

**Comment by EugeneFoo** [comment]
Hi Brandon,

I test with that however it still show me the same error. Even I delete payload it still shows the same error. I think this just happened 1-2 weeks ago, as I'm using exactly same code previously which is working fine.

**Comment by brandon-barker** [comment]
Okay thanks, I'll test your whole solution when I'm back on a Windows box later this afternoon :+1:

**Comment by EugeneFoo** [comment]
Thanks Brandon.

**Comment by brandon-barker** [comment]
Hi @EugeneFoo, sorry for the delay on this. Confirmed that this is an issue, it seems their API has since been changed - will push out a fix ASAP

**Comment by EugeneFoo** [comment]
Alright Thanks! 
Wait for your release then :)

**Comment by brandon-barker** [comment]
Hi @EugeneFoo, the PushBots team have gotten back to me on this issue and unfortunately they have deprecated this API endpoint due to performance issues... I have tried working around it but unfortunately I no longer have any way of retrieving devices from their API, so until they reinstate this endpoint this functionality will no longer work. :( 

**Edit**: to clarify, I mean the **GetDevices** method will no longer work.

Closing this issue until then, sorry for the inconvenience.
